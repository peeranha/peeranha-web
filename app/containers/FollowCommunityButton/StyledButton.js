import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import okayIcon from 'svg/okay';
import { blue, transparent } from 'style-constants';

import OutlinedButton from 'components/Button/OutlinedButton';
import Icon from 'components/Icon';

import Button from './index';
import messages from './messages';

/* istanbul ignore next */
const BStyled = OutlinedButton.extend`
  background-color: ${x => (x['data-isfollowed'] ? blue : transparent)};
  border: ${x => (x['data-isfollowed'] ? 'none' : '')};
`;

const B = /* istanbul ignore next */ ({ isFollowed, onClick }) => (
  <BStyled data-isfollowed={isFollowed} onClick={onClick}>
    {!isFollowed && <FormattedMessage {...messages.follow} />}

    {isFollowed && <Icon icon={okayIcon} noMargin />}
  </BStyled>
);

export const StyledButton = /* istanbul ignore next */ ({
  communityIdFilter,
}) => (
  <Button
    communityIdFilter={communityIdFilter}
    render={({ isFollowed, onClick }) => (
      <B isFollowed={isFollowed} onClick={onClick} />
    )}
  />
);

B.propTypes = {
  isFollowed: PropTypes.bool,
  onClick: PropTypes.func,
};

StyledButton.propTypes = {
  communityIdFilter: PropTypes.number,
};

export default React.memo(StyledButton);
