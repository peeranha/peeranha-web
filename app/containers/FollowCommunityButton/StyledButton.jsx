import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import okayIcon from 'images/okay.svg?inline';

import PrimaryButton from 'components/Button/Contained/PrimaryMedium';
import InfoButton from 'components/Button/Outlined/InfoMedium';

import Button from './index';
import messages from './messages';

const B = /* istanbul ignore next */ ({ isFollowed, onClick, id }) => {
  if (isFollowed) {
    return (
      <PrimaryButton id={id} data-isfollowed={isFollowed} onClick={onClick}>
        <img className="py-1" src={okayIcon} alt="icon" />
      </PrimaryButton>
    );
  }

  return (
    <InfoButton id={id} data-isfollowed={isFollowed} onClick={onClick}>
      <FormattedMessage {...messages.subscribe} />
    </InfoButton>
  );
};

export const StyledButton = /* istanbul ignore next */ ({
  communityIdFilter,
}) => (
  <Button
    communityIdFilter={communityIdFilter}
    render={({ isFollowed, onClick, id }) => (
      <B id={id} isFollowed={isFollowed} onClick={onClick} />
    )}
  />
);

B.propTypes = {
  isFollowed: PropTypes.bool,
  onClick: PropTypes.func,
  id: PropTypes.string.isRequired,
};

StyledButton.propTypes = {
  communityIdFilter: PropTypes.number,
};

export default React.memo(StyledButton);
