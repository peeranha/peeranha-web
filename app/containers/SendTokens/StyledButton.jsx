import styled from 'styled-components';

export default styled.button`
  position: relative;
`;