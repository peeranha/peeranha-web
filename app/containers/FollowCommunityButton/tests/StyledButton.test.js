import { StyledButton } from '../StyledButton';

describe('StyledButton', () => {
  it('test', () => {
    expect(StyledButton({ communityIdFilter: 1 })).toMatchSnapshot();
  });
});
