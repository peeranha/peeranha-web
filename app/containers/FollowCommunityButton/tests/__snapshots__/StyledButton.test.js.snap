// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton test 1`] = `
<withReducer(withSaga(Connect(FollowCommunityButton)))
  communityIdFilter={1}
  render={[Function]}
/>
`;
